web/src/components/tasks/search-business-name/SearchBusinessNameForm.tsx

```diff
@@ -51,42 +48,128 @@ export const SearchBusinessNameForm = (props: Props): ReactElement => {
 
   const { Config } = useConfig();
   const didInitialSearch = useRef<boolean>(false);
-  const { userData } = useUserData();
-  const [nameAvailability, setNameAvailability] = useState<NameAvailability | undefined>(undefined);
-  const [submittedName, setSubmittedName] = useState<string>("");
+  const { userData, updateQueue } = useUserData();
+  const {
+    setFormationFormData,
+    setBusinessNameAvailability,
+    setDbaBusinessNameAvailability,
+    setFieldsInteracted,
+  } = useContext(BusinessFormationContext);
+
+  const setNameAvailability = props.isDba ? setDbaBusinessNameAvailability : setBusinessNameAvailability;
 
+  const FIELD_NAME = "businessName";
   const SearchBusinessNameErrorLookup: Record<SearchBusinessNameError, string> = {
     BAD_INPUT: Config.searchBusinessNameTask.errorTextBadInput,
     SEARCH_FAILED: Config.searchBusinessNameTask.errorTextSearchFailed,
   };
 
-  useEffect(() => {
-    props.onChange && props.onChange(nameAvailability);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [nameAvailability]);
+  const emptyNameAvailability: NameAvailability = {
+    similarNames: [],
+    status: undefined,
+    lastUpdatedTimeStamp: "",
+  };
 
   const Unavailable = props.unavailable;
   const Available = props.available;
-  const onSubmit = props.onSubmit;
+
+  const handleBusinessNameAvailability = async (
+    nameAvailability: NameAvailability,
+    submittedName: string
+  ): Promise<void> => {
+    const validName = ["AVAILABLE", "UNAVAILABLE"].includes(nameAvailability.status ?? "");
+    const nameUnavailable = nameAvailability.status === "UNAVAILABLE";
+
+    if (updateQueue && validName) {
```

Comment:
We prefer guard clauses over if statements. The reason is to avoid having the "meat" of the internal logic of a function indented inwards... it should be as close to the top-level indentation as possible.

The difference between
```
const doThing = () => {
  if (thing) {
     doWhatever()
  }
}
```


```
const doThing = () => {
  if (!thing) return;
  doWhatever()
}
```

The reasoning here is that if you're skimming the doThing method and want to know what it does, then you can skim the top level and not have to search inwards to find that it calls doWhatever

Reply:
Fair point. Restructured.